### PR TITLE
Nemesis output with adaptivity + recover

### LINKED ISSUE #13396: Nemesis outputs are overwritten when running with recover
## Bug Description
While recovering a simulation with mesh adaptivity and nemesis outputs, the new output files override the previous ones instead of creating new files.  

## Steps to Reproduce
Run any test file with mesh adaptivity and output `nemesis=true`

## Impact
Recover does not work with nemesis and adaptivity.

### PR DESCRIPTION
**Pull request by permcody**
closes #13396

### CODE REVIEW

File: framework/src/outputs/Nemesis.C [hunk 1]
```diff
@@ -65,15 +64,8 @@ Nemesis::meshChanged()
...
-  _file_num++;
-
-  // Reset the number of outputs for this file
-  _nemesis_num = 1;
-
-  // Create the new NemesisIO object
-  _nemesis_io_ptr = libmesh_make_unique<Nemesis_IO>(_problem_ptr->mesh().getMesh());
-  _nemesis_initialized = false;
+  // Indicate to the Exodus object that the mesh has changed
+  _nemesis_mesh_changed = true;
```
Comment (SudiptaBiswas) [inline framework/src/outputs/Nemesis.C]: Should not this be set within `meshChanged()`?

File: test/tests/outputs/nemesis/tests [hunk 2]
```diff
@@ -71,9 +71,39 @@
...
+    min_parallel = 4
+    max_parallel = 4
+    mesh_mode = distributed
+    # We require a recent version of PETSc so that a version of METIS/ParMETIS with a portable RNG is used.
+    petsc_version = '>=3.8.0'
+    design = 'Outputs/index.md'
+    issues = '#2122'
+    requirement = "The Nemesis Output object shall support writing separate files per processor during a transient simulation."
+    recover = false
+  [../]
```
Comment (SudiptaBiswas) [inline test/tests/outputs/nemesis/tests]: Should not this be tested with recover?

File: test/tests/outputs/nemesis/tests [hunk 3]
```diff
@@ -71,9 +71,39 @@
...
+    type = 'Exodiff'
+    input = nemesis_transient.i
+    exodiff = 'nemesis_transient_out.e-s0010.4.0 nemesis_transient_out.e-s0010.4.1 nemesis_transient_out.e-s0010.4.2 nemesis_transient_out.e-s0010.4.3'
+    min_parallel = 4
+    max_parallel = 4
+    mesh_mode = distributed
+    # We require a recent version of PETSc so that a version of METIS/ParMETIS with a portable RNG is used.
+    petsc_version = '>=3.8.0'
+    design = 'Outputs/index.md'
+    issues = '#2122'
```
Comment (SudiptaBiswas) [inline test/tests/outputs/nemesis/tests]: Issue #13396 to be added.

**Comment by frombs** [comment]
I have a large simulation with Nemesis output that I am trying to restart using the --recover option.  When the recovered simulation gets to the start of "Outputing nemesis",  it terminates with exit code 11.  Looking at the "cpr/xdr" files, none of them were appended or overwritten, and no new files were created. I am hoping this PR will resolve the issue.  I see that it went stale a couple week ago.  What needs to be done to finalize it?  By the way, I am not using adaptivity so it may be a different problem.

**Comment by SudiptaBiswas** [comment]
Currently, nemesis output is not compatible with recover. This PR should fix that, with or without adaptivity.

**Comment by SudiptaBiswas** [comment]
@permcody Let me know if I can do anything to get this in.

**Comment by frombs** [comment]
I added the changes from commit 9610562bce04b4706e677421cdc41eec08767ae6 to my stack on HPC and tried the --recover option again.  It gave me the same exit code 11.  Will the changes only work going forward?  In other words, will existing recovery files remain broken and not usable?

**Comment by permcody** [comment]
@frombs - generally checkpoint format is not guaranteed to work across different versions of the code base.

**Comment by permcody** [comment]
There are several problems with the nemesis format. It appears that some of the support that was added for appending and such on the Exodus side has been actively used and given the same level of support on the Nemesis side. I think this PR is an incremental improvement what we have today but as @SudiptaBiswas pointed out. It still has significant issues. It'll take some time to get full recover support going for this format. I know that we'll need a few patches on the libMesh side to get this working completely. I'd be fine re-opening and getting this version in. I think a lot more work is needed though.

**Comment by permcody** [comment]
BTW - I don't have time to work on this. My July is pretty much booked solid with other critical work and August is starting to fill for me as well. Our resources are very low for this kind of stuff. That being said, I know that this may hold up some of Sudipta's work so I'd like a hackish solution to make it in there so that she can at least run. I was hoping that this version might be good enough to keep going for now.

**Pull request closed without merging**